### PR TITLE
Prevent unauthorized access to Patient resource before it happens

### PR DESCRIPTION
**Pull request by lmsurpre**
This prevents us from leaking information about what patient resources
exist / don't exist on the system, which was deemed useful in case an
implementer uses sensitive data for the Resource.id of the Patient
resources (which still isn't advised).

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

**Review by michaelwschroeder — APPROVED** [review]

**Comment by michaelwschroeder** [comment]
LGTM

**Review by JohnTimm — APPROVED** [review]
LGTM

File: fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java [hunk 1]
```diff
@@ -47,6 +47,19 @@
     private static final String BEARER_TOKEN_PREFIX = "Bearer";
     private static final String PATIENT_REF_PREFIX = "Patient/";
 
+    private static final String REQUEST_NOT_PERMITTED = "Requested interaction is not permitted by any of the passed scopes.";
+
+    @Override
+    public void beforeRead(FHIRPersistenceEvent event) throws FHIRPersistenceInterceptorException {
```
Comment (lmsurpre) [inline fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java:53]: expand to beforeVread and beforeHistory as well

Comment (lmsurpre) [reply fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java:53]: done

**Review by d0roppe — COMMENTED** [review]
LGTM from what I understand.  Message is consistent?

**Comment by lmsurpre** [comment]
> Message is consistent?

Since we're not divulging any information that the user hasn't directly passed us, I opted for a more verbose message.  Example:
> Interaction with 'Patient/123abc' is not permitted under patient context 'xyz'.

This contrasts with the more generic message we return in the other cases where the resource(s) involved in the interaction belong to some compartment that we (and potentially the end user) doesn't know until it is executed.

**Pull request merged**